### DISCUSSION TITLE
Graphql Dynamic Client failed due to missing field

### DISCUSSION THREAD

**Discussion by hantsy** [quarkusio/quarkus#40393] [Q&A]
Today I try to clean up my example projects: https://github.com/hantsy/quarkus-sandbox/tree/master/graphql-client

Given Graphql type:

```java
public record Post(
        String id,
        String title,
        String content,
        int countOfComments,
        List<Comment> comments
) {

}
```

And I created a dynamic client:

```java
@ApplicationScoped
public class PostDynamicClient {

    @Inject
    @GraphQLClient("post-client-dynamic")
    DynamicGraphQLClient dynamicClient;

    public List<Post> getAllPosts() throws ExecutionException, InterruptedException {
        Document query = document(
                operation(
                        field("allPosts",
                                field("id"),
                                field("title"),
                                field("content"),
                               // field("countOfComments"),
                                field("comments",
                                        field("id"),
                                        field("content")
                                )
                        )
                )
        );
        Response response = dynamicClient.executeSync(query);
        return response.getList(Post.class, "allPosts");
    }
}
```

I used the example project: https://github.com/hantsy/quarkus-sandbox/tree/master/graphql to serve graphql service.

When running the project: `mvn clean package quarkus:run`. It throws an exception: `missing countOfComment field`.

I have to uncomment the above line `// field("countOfComments"),` to make it work.

But for the client, it should assemble the fields freely. 

**The original codes(without countOfComments field, and `Post` is a POJO bean instead of the current record class) was also working well when I wrote down the examples.**

[a comment by an automated account is not shown]

**Comment by jmartisk** [comment]
That's because `countOfComments` in your `Post` record is a primitive `int`, and primitives are always assumed to be required (non-nullable) and so when you run a request that doesn't fetch this field, it will fail. It should work when you change the field to `Integer`, then the field will be considered nullable

**Reply by hantsy** [reply]
@jmartisk Changed to `Integer`, it works again.

But the original `int` also worked in the old quarkus.
